adjusted math.h

Zombieanfuehrer opened the pull request:
enhanced libm.h with missing declarations
added architecture-specific alltypes.h differences

libm.h - Added missing function declarations and weak_alias macro

alltypes.h - ARM v7 type definitions
fp_arch.h - ARM v7 floating-point assembly barriers
atomic_arch.h - Atomic operations (copied)
fenv.h - Floating-point environment (copied)

Kopfhoerer00 committed:
adjusted math.h

eleir9268 commented on fenv/armle-v7/bits/fenv.h:
This project is a "fork" of musl at https://git.musl-libc.org/cgit/musl (latest) with some features removed. I suggest copying arm support from there.

This file should be:
```c
#ifndef __ARM_PCS_VFP
#define FE_ALL_EXCEPT 0
#define FE_TONEAREST  0
#else
#define FE_INVALID    1
#define FE_DIVBYZERO  2
#define FE_OVERFLOW   4
#define FE_UNDERFLOW  8
#define FE_INEXACT    16
#define FE_ALL_EXCEPT 31
#define FE_TONEAREST  0
#define FE_DOWNWARD   0x800000
#define FE_UPWARD     0x400000
#define FE_TOWARDZERO 0xc00000
#endif

typedef unsigned long fexcept_t;

typedef struct {
	unsigned long __cw;
} fenv_t;

#define FE_DFL_ENV      ((const fenv_t *) -1)
```

Zombieanfuehrer replied:
Thanks for pointing out the official arm support eleir9268.

I have adjusted fenv/armle-v7/bits/fenv.h accordingly.

eleir9268 replied:
There should be a fenv/armle-v7/fenv.c and fenv/armle-v7/fenv-hf.S file providing definitions for the symbols in fenv/fenv.h

Zombieanfuehrer replied:
Definitions for the symbols throught adding the headers solved

Zombieanfuehrer committed:
Update fenv.h

adjusted cause review comment https://github.com/qnx-ports/muslflt/pull/2#discussion_r2231693053

eleir9268 commented on math/armle-v7/atomic_arch.h:
This file should be a copy of musl/arch/arm/atomic_arch.h at https://git.musl-libc.org/cgit/musl

Zombieanfuehrer replied:
I have made the further adjustments, thanks for the support!
see commid id 9c156fc30c8fd8e16ca6badef987fce1514d87dd

eleir9268 reviewed (commented):
@Zombieanfuehrer If you want this to be more expedient I can take over merging these changes, just let me know. Thanks

Zombieanfuehrer committed:
Add files via upload

added definitions for the symbols in fenv/fenv.h

Kopfhoerer00 committed:
customizations for qnx ports MR

Kopfhoerer00 committed:
Merge branch 'feature/add_qnx71_armv7le_sup' of https://github.com/Zombieanfuehrer/muslflt into feature/add_qnx71_armv7le_sup

eleir9268 commented on math/armle-v7/atomic_arch.h:
I think all that's missing is the ISA specific definitions for math functions, defined by upstream in,
```
src/math/arm/fabs.c
src/math/arm/fabsf.c
src/math/arm/fma.c
src/math/arm/fmaf.c
src/math/arm/sqrt.c
src/math/arm/sqrtf.c
```
which should only impact performance. After these files are added I can manually test these changes against abseil's test suite.

jaume-pladevall-itk commented:
Hello, I don't know if this is the best place to add this, but on my side I had to rename the folder armle-v7 to armle to go past the compilation errors related to math and fenv not being found. 
When running: 
`QNX_PROJECT_ROOT="$(pwd)/muslflt"` make -C build-files/ports/muslflt/ INSTALL_ROOT_nto=<staging-install-folder> USE_INSTALL_ROOT=true install -j4`

jaume-pladevall-itk commented on math/armle-v7/atomic_arch.h:
```diff
@@ -0,0 +1,107 @@
+#define BLX "blx"
+#endif
+
+extern hidden uintptr_t __a_cas_ptr, __a_barrier_ptr;
```
I had to include "features.h" in this file, because I was getting an error saying _hidden_ was an unknown type.
